src/BondEvaluatorDoubleWell.h

```diff
@@ -2,108 +2,95 @@
 
+namespace hoomd
+    {
 namespace azplugins
     {
 namespace detail
     {
 
-//! Class for evaluating the double well bond potential
-/*!
-This bond potential follows the functional form
-\f{eqnarray*}
-
-V_{\rm{DW}}(r)  =   \frac{V_{max}-c/2}{b^4} \left[ \left( r - a/2 \right)^2 -b^2 \right]^2 +
-\frac{c}{2b}\left(r-a/2\right)+c/2
-
-\f}
-which has two minima at r = (a/2 +/- b), seperated by a maximum at a/2 of height V_max when c is set
-to zero.
-
-The parameter a tunes the location of the maximal value and the parameter b tunes the distance of
-the two maxima from each other.  This potential is useful to model bonds which can be either
-mechanically or thermally "activated" into a effectively longer state. The value of V_max can be
-used to tune the height of the energy barrier in between the two states.
-
-If c is non zero, the relative energy of the minima can be tuned, where c is the energy of the
-second minima, the first minima value is at zero. This  causes a small shift in the location of the
-minima and the maxima, because of the added linear term.
-
-The parameters are:
-    - \a V_max (params.x) potential difference between the the first minima and maxima
-    - \a a (params.y) shift for the location of the V_max, the maximum is at approx. a/2
-    - \a b (params.z) scaling for the distance of the two minima at approx. (a/2 +/- b)
-    - \a c (params.w) potential difference between the two minima, default is c=0
-
-*/
-class BondEvaluatorDoubleWell
+//! Parameters of double well bond potential
+struct BondParametersDoubleWell : public BondParameters
     {
-    public:
-    //! Define the parameter type used by this bond potential evaluator
+#ifndef __HIPCC__
+    BondParametersDoubleWell() : V_max(0), a(0), b(0), c(0) { }
 
-    typedef Scalar4 param_type;
-
-    //! Constructs the pair potential evaluator
-    /*!
-     * \param _rsq Squared distance beteen the particles
-     * \param _params Per type bond parameters of this potential as given above
-     */
-    DEVICE BondEvaluatorDoubleWell(Scalar _rsq, const param_type& _params)
-        : rsq(_rsq), V_max(_params.x), a(_params.y), b(_params.z), c(_params.w)
+    BondParametersDoubleWell(pybind11::dict v)
         {
+        V_max = v["V_max"].cast<Scalar>();
+        a = v["a"].cast<Scalar>();
+        b = v["b"].cast<Scalar>();
+        c = v["c"].cast<Scalar>();
         }
 
-    //! This evaluator doesn't use diameter information
-    DEVICE static bool needsDiameter()
+    pybind11::dict asDict()
         {
-        return false;
+        pybind11::dict v;
+        v["V_max"] = V_max;
+        v["a"] = a;
+        v["b"] = b;
+        v["c"] = c;
+        return v;
         }
+#endif
 
-    //! Accept the optional diameter values
-    /*!
-     * \param da Diameter of particle a
-     * \param db Diameter of particle b
-     */
-    DEVICE void setDiameter(Scalar da, Scalar db) { }
+    Scalar V_max; //!< Potential difference between the the first minima and maxima
```

Comment:
Naming suggestion: 
`V_max` -> `V_barrier` = Potential barrier height between the two minima as measured between the first minimum and maximum 
`a/2` -> `r_tr` location of the transition state where the potential barrier is maximum
`b` ->  `r_w` width of the double well potential 
`c` -> `V_a`  Potential height in the activated state with respect to the first minimum 

Honestly, it might make sense to redefine the parameters all-together. `V_max, a,b,c` are the "native" variables going into the potential, but it might be more user friendly to define the location of the first minimum (`a/2-b`), and `a/2` (location of the transition state) instead of `a` and `b`. Then we can calculate the necessary parameters from the user input on the python level. But it is also ok as is, a matter of taste/preference...

Reply:
`V_a` could also be `V_tilt` or `V_offset`...

Reply:
I am good with whatever you like here, since this is mostly your potential! If you can make a recommendation, we can implement it.

Reply:
I thought about this and:
* I would rename the variables based on the minimum and transition state _only if_ we remove the `c` parameter. I think `c` makes these values change, right?
* If we don't remove the `c` parameter, I think we should just leave things more or less as is, and we can add some documentation suggesting how to choose them
@astatt let me know which you prefer so we can finalize this today.